src/main/java/io/cdap/plugin/http/common/BaseHttpConfig.java

```diff
@@ -157,6 +159,12 @@ public abstract class BaseHttpConfig extends ReferencePluginConfig {
     @Macro
     protected String serviceAccountScope;
 
+    /**
+     * The credential object to help with getting the access token
+     */
+    @Nullable
+    private OAuth2Credentials credentials;
```

Comment:
the config class defines configuration properties that the user can supply, it is not meant for putting arbitrary variables of different types.